index.js

```diff
@@ -37,19 +38,33 @@ export default async function pWaitFor(condition, options = {}) {
 		}
 	});
 
-	if (timeout !== Number.POSITIVE_INFINITY) {
-		try {
-			return await pTimeout(promise, timeout);
-		} catch (error) {
-			if (retryTimeout) {
-				clearTimeout(retryTimeout);
-			}
+	if (timeout === Number.POSITIVE_INFINITY) {
+		return promise;
+	}
 
-			throw error;
+	const milliseconds = timeout;
+
+	const {
+		message: timeoutMessage,
+		milliseconds: timeoutMs,
+		customTimers
+	} = typeof timeout === 'number' ? {milliseconds} : timeout;
+
+	const customTimersOptions = customTimers ? {customTimers} : undefined;
+
+	try {
+		return await pTimeout(promise, {
+			milliseconds: timeoutMs,
+			message: timeoutMessage,
+			customTimers: customTimersOptions
```

Comment:
Isn’t this wrong? The object will be `customTimers.customTimers`. I don’t think you need the `customTimersOptions` at all. 
```suggestion
			customTimers
```